html/semantics/embedded-content/media-elements/playing-the-media-resource/playbackRate.html

```diff
@@ -9,11 +9,46 @@
   assert_equals(v.playbackRate, 1);
 }, 'playbackRate initial value');
 
-async_test(function(t) {
+function testPlaybackRateHelper(t, newPlaybackRate) {
   var v = document.createElement('video');
-  v.playbackRate = 2;
-  v.addEventListener('ratechange', t.step_func(function() {
+  var initialRate = v.playbackRate;
+
+  v.addEventListener('ratechange', function() {
```

Comment:
Oh, wait, there's a slightly better way to do this. If you do

```js
v.addEventListener('ratechange', t.step_func(function() {
  ...
  t.done();
}));
```

then exceptions thrown will properly be redirected to the test harness. That should cause Firefox to fail the test, instead of timing out.

You can then make it even shorter with an extra shortcut:

```js
v.addEventListener('ratechange', t.step_func_done(function() {
  ...
}));
```